Multisig/Data/Services/Safe Transaction Service/SignTransactionRequest.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SignTransactionRequest.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 17.09.20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Web3
+
+struct SignTransactionRequest: JSONRequest {
+    let safe: String
+
+    let to: String?
+    let value: String
+    let data: String?
+    let operation: Int
+    let safeTxGas: String
+    let baseGas: String
+    let gasPrice: String
+    let gasToken: String?
+    let refundReceiver: String
+    let nonce: String
+    let contractTransactionHash: String
+    let sender: String
+    let signature: String?
+
+    enum CodingKeys: String, CodingKey {
+        case sender, to, nonce, safeTxGas, baseGas, gasPrice, gasToken, refundReceiver,
+             contractTransactionHash, data, operation, value, signature
+    }
+
+    enum RequestError: String, Error {
+        case couldNotVerifySafeTxHash
+    }
+
+    init(transaction: Transaction, safeAddress: Address) throws {
+        safe = safeAddress.checksummed
+        to = transaction.to?.description
+        value = transaction.value.description
+        data = transaction.data?.description
+        operation = transaction.operation.rawValue
+        safeTxGas = transaction.safeTxGas.description
+        baseGas = transaction.baseGas.description
+        gasPrice = transaction.gasPrice.description
+        gasToken = transaction.gasToken?.description
+        refundReceiver = transaction.refundReceiver.description
+        nonce = transaction.nonce.description
+        contractTransactionHash = transaction.safeTxHash.description
+
+        let hashToSign = Data(ethHex: contractTransactionHash)
+
+        guard let pkData = try? App.shared.keychainService.data(forKey: KeychainKey.ownerPrivateKey.rawValue),
+              let pk = try? EthereumPrivateKey(pkData.bytes),
+              let sig = try? pk.sign(hash: hashToSign.bytes) else {
+            sender = ""
```

Comment:
I see that the errors are not handled here, why is that? I.e. if key is invalid or missing, we'll send the request that will be failing but we probably won't know the actual reason

Reply:
I think this should be a separate task to properly manage missing private key (like when the app is recovered from the iCloud)
wdyt?

Reply:
I  think this is not needed at the moment, especially that there was decision not to do iCloud syncing. This code block just skips errors. While  this is ok during  prototyping, this is not ready for production for, in my opinion. Throwing and  showing human-readable message to the user would be better.

Reply:
I mean when the user changes the phone. We could have an address stored, but not the PK, that could lead to a confusion.
I see signing might fail only in case of this misconfiguration